pale/adapters/webapp2.py

```diff
@@ -72,27 +72,26 @@ class DefaultWebapp2Context(pale.context.DefaultContext):
 
     def build_args_from_request(self, request):
         keys = request.arguments()
-        args = {}
-
-        j_args = self.deserialize_args_from_body(request.body)
+        req_args = {}
 
         for key in keys:
-            args[key] = request.get_all(key)
+            req_args[key] = request.get_all(key)
 
-        # json/body args
-        for k,v in j_args.iteritems():
-            if k in args:
-                logging.warning("Overwriting request arg %s with json arg",
-                        k)
-            args[k] = v
+        if request.content_type == 'application/json':
+            json_args = request.json_body
```

Comment:
@soundofjw Have a gander here.  i'm guessing most http libraries will already have some kind of functionality like this, so no sense in duplicating the actual deserialization.